superset-frontend/src/SqlLab/components/SqlEditor.jsx

```diff
@@ -613,6 +630,9 @@ class SqlEditor extends React.PureComponent {
               isRunning={this.props.latestQuery.state === 'running'}
             />
           )}
+          <Dropdown overlay={this.renderDropdown()} arrow>
```

Comment:
It looks like `Dropdown` takes either a Node or a method, so you could pass it just the method as the overlay, and leave it up to ant to decide when to rerun the method. Otherwise, I think it's going to run on each re-render.